lib/instance.js

```diff
@@ -45,18 +59,20 @@ module.exports = function instance(input, done) {
     // calculate driver configuration
     driverConfig = profileConf.driver;
     config.set('driver', Object.assign({}, config.get('driver'), driverConfig));
-    runner = mocha.run(function (failures) {
-      // only attach this listener if we aren't running in parallel
-      if (!done && process) {
-        process.on('exit', function () {
-          // exit with non-zero status if there were failures
-          process.exit(failures);
-        });
-      }
-      // attempted to run with no matching suites/tests
-      if (!anyTests && done) {
-        done(result);
+
+    if (profileConf.parallel === 'file') {
+      // if we're running parallel by file, make sure we really need to run()
+      mocha.loadFiles();
```

Comment:
We must load files for the `runner.grep()` to have something to match against in the `mocha.suite`. [`mocha.loadFiles()` was always being called in `mocha.run()`](https://github.com/mochajs/mocha/blob/2bc9c4d1d638c8a95250757dd23724f788918abe/lib/mocha.js#L513), and I'm nearly certain that there is no consequence of running it a few lines earlier.